cmd/scaler/scaler.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	appsV1 "k8s.io/api/apps/v1"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/prombench/pkg/provider/k8s"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type scale struct {
+	k8sClient *k8s.K8s
+	min       int32
+	max       int32
+	interval  time.Duration
+}
+
+func new() *scale {
+	k, err := k8s.New(context.Background(), nil)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, errors.Wrapf(err, "Error creating k8s client inside the k8s cluster"))
+		os.Exit(2)
+	}
+	return &scale{
+		k8sClient: k,
+	}
+}
+
+func (s *scale) updateReplicas(replicas *int32) []k8s.Resource {
+	var k8sResource []k8s.Resource
+	for _, deployment := range s.k8sClient.K8sResource {
+		k8sObjects := make([]runtime.Object, 0)
+
+		for _, resource := range deployment.Objects {
+			if kind := strings.ToLower(resource.GetObjectKind().GroupVersionKind().Kind); kind == "deployment" {
+				req := resource.(*appsV1.Deployment)
+				req.Spec.Replicas = replicas
+				k8sObjects = append(k8sObjects, req)
+			}
+		}
+		if len(k8sObjects) > 0 {
+			k8sResource = append(k8sResource, k8s.Resource{FileName: deployment.FileName, Objects: k8sObjects})
+		}
+	}
+	return k8sResource
+}
+
+func (s *scale) scale(*kingpin.ParseContext) error {
+	log.Printf("Starting Prombench-Scaler:\n\t min: %d\n\t max: %d\n\t interval: %s", s.min, s.max, s.interval)
+
+	minResourceObjects := s.updateReplicas(&s.min)
+	maxResourceObjects := s.updateReplicas(&s.max)
+
+	for {
+		time.Sleep(s.interval)
+
+		log.Printf("Scaling Deployment to %d", s.min)
+		if err := s.k8sClient.ResourceApply(minResourceObjects); err != nil {
+			fmt.Fprintln(os.Stderr, errors.Wrapf(err, "Error scaling deployment"))
+			os.Exit(2)
+		}
+
+		time.Sleep(s.interval)
+
+		log.Printf("Scaling Deployment to %d", s.max)
+		if err := s.k8sClient.ResourceApply(maxResourceObjects); err != nil {
+			fmt.Fprintln(os.Stderr, errors.Wrapf(err, "Error scaling deployment"))
+			os.Exit(2)
+		}
+	}
+}
+
+func main() {
+
+	app := kingpin.New(filepath.Base(os.Args[0]), "The Prombench-Scaler tool")
+	app.HelpFlag.Short('h')
+
+	s := new()
+
+	k8sApp := app.Command("scale", `Scale a Kubernetes deployment object periodically up and down`).
```

Comment:
can we have a full example somewhere. don't forget to test locally that is shows as expected when typing a wrong command or scaler -h. I think prombench was done with some examples.

When we have full examples in the README.md can just have: run `scaler -h` for usage and examples. 
And no need to maintain the README.md when we update the usage and commands.

Reply:
One issue is when an in-cluster k8s clientset is created https://github.com/sipian/prombench/blob/refactor-loadgen-scaler/cmd/scaler/scaler.go#L28.

If the working environment is not inside a cluster, we get the error
 `Error creating k8s client inside the k8s cluster: k8s config error: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined
`

This error even comes for `./scaler -h` .
Even if we separate this into a separate Action, this error message will still be there.